wizard/src/utils/detect_location.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { DetectedDetails, LocationFormat, ConfidenceLevel } from "../types";
+import countriesJSON from './country_mappings.json';
+
+// A LocationDetector objected is meant to be initialized once. It provides
+// convenience access to all location types and their supported formats. It also
+// supports detecting the location type for each individual column (a header and
+// a list of string values).
+export class LocationDetector {
+  countryNames: Set<string>;
+  countryISO: Set<string>;
+  countryAbbrv3: Set<string>;
+  countryNumeric: Set<string>;
+
+  static typeFormatMappings = new Map<string, Array<LocationFormat>>(
+    [
+      ['None',        [ {propertyName: "name", displayName: "Full Name"},]],
+      ['LatLon',      [ {propertyName: "name", displayName: "Full Name"},]],
+      ['Country',     [ {propertyName: "name", displayName: "Full Name"},
+                        {propertyName: "iso", displayName: "ISO Code"},
```

Comment:
Match property names to what we have in BQ?

Namely:
iso -> isoCode
abbrv3 -> countryAlpha3Code
numeric - > countryNumericCode

Reply:
We can use constants for this, so the in-memory version of the JSON can be a map from property -> List... for future PR.

Reply:
For now, I have made the string replacements. Happy to improve in the future PR?

Reply:
Absolutely!  Thanks.